src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -143,11 +143,19 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 
 		if (this.messagesStore.selectedMessageId) {
 			try {
-				this.messageAC = new AbortController();
-				message = await this.api.messages.getMessage(
-					this.messagesStore.selectedMessageId.valueOf(),
-					this.messageAC.signal,
-				);
+				if (
+					await this.api.messages.matchMessage(
+						this.messagesStore.selectedMessageId.valueOf(),
```

Comment:
this.messagesStore.selectedMessageId is still used to fetch messageIds even if doesnt match the filter which is probably wrong.
so i guess we need to reset it here